Data_Structures_Answers.md

```diff
@@ -1,18 +1,25 @@
 Add your answers to the questions below.
 
 1. What is the runtime complexity of your ring buffer's `append` method?
+O(1)
 
 2. What is the space complexity of your ring buffer's `append` function?
+o(1)
 
 3. What is the runtime complexity of your ring buffer's `get` method?
+O(n)
 
 4. What is the space complexity of your ring buffer's `get` method?
-
+O(n)
 
 5. What is the runtime complexity of the provided code in `names.py`?
+O(n^2)
 
 6. What is the space complexity of the provided code in `names.py`?
+O(n^2)
 
 7. What is the runtime complexity of your optimized code in `names.py`?
+O(n)
 
 8. What is the space complexity of your optimized code in `names.py`?
+O(1)
```

Comment:
Since this is creating two new sets in memory at some point in the operations, this will end up being O(n) space complexity.